Assets/GameEngine.cs

```diff
@@ -6,17 +6,35 @@ public class GameEngine : MonoBehaviour {
     private Transform source;
 
 	public void RegisterStrongholdClick(Transform selected) {
-        if (source == null) {
+
+        ObjectOwnerState selectedInfo = selected.GetComponent<ObjectOwnerState>();
+
+        if (source == null && selectedInfo.player!=null) {
```

Comment:
Great stuff 💯 Nitpick: Would add spaces so it's like
```
selectedInfo.player != null
```

Could you also update the comment after this line and move it before it? I think it would make sense if it's before this.